watchtower-core/src/main/java/com/tracelink/appsec/watchtower/core/rule/RuleService.java

```diff
@@ -158,16 +158,18 @@ public boolean createsNameCollision(long id, String name) {
 			RuleEntity found = ruleRepository.findByName(rule.getName());
 			if (found != null) {
 				if (rule.getRuleDesignation().equals(RuleDesignation.PROVIDED)) {
+					ProvidedRuleDto provided = (ProvidedRuleDto) rule;
```

Comment:
Do you need this cast? All of the methods used here are also on `RuleDto`

Reply:
I just thought it was cleaner. you're correct that it's not required. what's your take? cleaner? or unnecessary?

Reply:
I'm leaning towards unnecessary but I'll leave it up to you. It's really a small detail anyway